source/agora/consensus/protocol/Nominator.d

```diff
@@ -116,7 +117,7 @@ extern(D):
         this.params = params;
         this.clock = clock;
         this.network = network;
-        this.key_pair = key_pair;
+        this.key_pair = Pair.fromScalar(secretKeyToCurveScalar(key_pair.secret));
```

Comment:
Can we just change the argument instead ?

Reply:
We need the address from the keyPair in the next line. (Assuming you mean the Nominator constructor arg key_pair).

Reply:
@Geod24 I think I asked this once before but I don't recall getting an answer: can't we get rid of `PrivateKey` / `PublicKey` and just use Schnorr? Or is there some compatibility we need to keep around?